flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -651,39 +681,52 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
         }
     }
 
+    private CompletableFuture<CheckpointPlan> calculateCheckpointPlan() {
+        return checkpointPlanCalculator
+                .calculateCheckpointPlan()
+                // Disable checkpoints after tasks finished according to the flag.
+                .thenApplyAsync(
+                        plan -> {
+                            if (!allowCheckpointsAfterTasksFinished
+                                    && !plan.getFinishedTasks().isEmpty()) {
+                                throw new CompletionException(
```

Comment:
Should we do this check before computing the plan? 
Otherwise, 
1. the flag doesn't prevent existing deployments from potential performance degradation
2. the plan is computed for no purpose

I think it can also be put it planCalculator if it's easier.

WDYT?

Reply:
I also think move the check before computing would be more reasonable. I have modified accordingly.